src/rpc/methods/eth.rs

```diff
@@ -1282,6 +1293,49 @@ impl RpcMethod<2> for EthGetBlockByNumber {
     }
 }
 
+pub enum EthGetBlockTransactionCountByNumber {}
+impl RpcMethod<1> for EthGetBlockTransactionCountByNumber {
+    const NAME: &'static str = "Filecoin.EthGetBlockTransactionCountByNumber";
+    const PARAM_NAMES: [&'static str; 1] = ["block_number"];
+    const API_VERSION: ApiVersion = ApiVersion::V1;
+    const PERMISSION: Permission = Permission::Read;
+
+    type Params = (Int64,);
+    type Ok = Uint64;
+
+    async fn handle(
+        ctx: Ctx<impl Blockstore + Send + Sync + 'static>,
+        (block_number,): Self::Params,
+    ) -> Result<Self::Ok, ServerError> {
+        let height = block_number.0;
+        let head = ctx.chain_store.heaviest_tipset();
+        if height > head.epoch() {
+            return Err(anyhow::anyhow!("requested a future epoch (beyond \"latest\")").into());
+        }
+        let ts = ctx.chain_store.chain_index.tipset_by_height(
+            height,
+            head,
+            ResolveNullTipset::TakeOlder,
+        )?;
+        let count = count_messages_in_tipset(ctx.store(), &ts)?;
+        Ok(Uint64(count as _))
+    }
+}
+
+fn count_messages_in_tipset(store: &impl Blockstore, ts: &Tipset) -> anyhow::Result<usize> {
+    let mut message_cids = CidHashSet::default();
+    for block in ts.block_headers() {
+        let (bls_messages, secp_messages) = crate::chain::store::block_messages(store, block)?;
```

Comment:
Can messages in a block be duplicated?

Reply:
In one block I think no, in N blocks yes